sklearn/linear_model/logistic.py

```diff
@@ -16,7 +16,8 @@
 import numpy as np
 from scipy import optimize, sparse
 
-from .base import LinearClassifierMixin, SparseCoefMixin, BaseEstimator
+from .base import (LinearClassifierMixin, SparseCoefMixin, BaseEstimator,
+                   LinearModel)
```

Comment:
I suppose that ``LogisticRegression`` was intended to extend the ``LinearModel`` here to use ``_preprocess_data`` and ``_set_intercept``. I am not sure about the validity of the changes, but the tests are failing due to this.